src/mediasession-helper.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {tryParseJson} from './json';
+
+/**
+ * Updates the Media Session API's metadata
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ * @param {!./video-interface.VideoMetaDef} metaData
+ * @param {function} playHandler
+ * @param {function} pauseHandler
+ */
+export function setMediaSession(ampdoc,
+                                metaData,
+                                playHandler = null,
+                                pauseHandler = null) {
+  const win = ampdoc.win;
+  const navigator = win.navigator;
+  if ('mediaSession' in navigator && win.MediaMetadata) {
+    // Clear mediaSession (required to fix a bug when switching between two
+    // videos)
+    navigator.mediaSession.metadata = new win.MediaMetadata({
+      title: '',
+      artist: '',
+      album: '',
+      artwork: [
+        { src: ''},
+      ]
+    });
+    // Add metaData
+    navigator.mediaSession.metadata = new win.MediaMetadata(metaData);
```

Comment:
nit: do a sweep of variable naming for this PR so we always do `(M/m)etadata` instead of `(M/m)etaData` to match native naming.